### PR TITLE
JS linting and building

### PR DESCRIPTION
**Pull request by vhf**
I read a bit of the JS front-end codebase and noticed a few inconsistencies, then I noticed the gulp linting pipeline wasn't checking anything because it lacked proper configuration. I went for the angular code style, it's very close to the existing JS code and didn't require many changes for the JS linting step to pass.
- a33b3e9: gulp global installation takes a while and there is no reason to reinstall it at
  every gulp build
- 2bf4083: By default jscs doesn't enforce anything. Add a `.jscsrc` based on the [angular
  style guide](https://github.com/johnpapa/angular-styleguide/tree/master/a1#jscs) endorsed by the angular team.

### CODE REVIEW

File: orchestra/static/orchestra/common/components/checklist/js/directives.js [hunk 1]
```diff
@@ -21,7 +21,7 @@
           'comments': [{
             'author': 'jane',
             'timestamp': new Date(new Date() - 3 * 60 * 60 * 1000),
-            'text': "The hero image you've chosen doesn't fit " +
+            'text': 'The hero image you\'ve chosen doesn\'t fit ' +
```
Comment (marcua) [inline orchestra/static/orchestra/common/components/checklist/js/directives.js]: Our Python style guide suggests we do single quotes unless we'd have to escape, and then we do double quotes.  Any way to encode this in the `.jscsrc`?  If not, then such is life :)

**Comment by marcua** [comment]
What a nice contribution @vhf !  Thank you for submitting it.  I left one question, and invited @jrbotros to take a look as well.

For some reason our CircleCI build failed on this (https://circleci.com/gh/b12io/orchestra/1457) but it might be transient.  Let's see if rebuilding helps.

File: orchestra/static/orchestra/common/components/checklist/js/directives.js (hunk 1 above)
Comment (vhf) [inline orchestra/static/orchestra/common/components/checklist/js/directives.js]: Sure, I added `"validateQuoteMarks": { "mark": "'", "escape": true },` to the rules and reverted the relevant changes!

**Comment by vhf** [comment]
Thanks! The CircleCI issue was mentioned [here](https://github.com/b12io/orchestra/pull/348#issuecomment-238665906) the other day, I guess it's still a coveralls issue.

**Comment by marcua** [comment]
lgtm!  thank you so much for your contribution!

**Pull request merged**